src/EventStore.Core/TransactionLog/Chunks/ChunkHeader.cs

```diff
@@ -39,36 +45,66 @@ public class ChunkHeader {
 			ChunkEndPosition = (ChunkEndNumber + 1) * (long)ChunkSize;
 		}
 
+		public ChunkHeader(ReadOnlySpan<byte> source) {
+			Debug.Assert(source.Length >= Size);
+
+			SpanReader<byte> reader = new(source);
+
+			if ((FileType)reader.Read() is not FileType.ChunkFile)
+				throw new CorruptDatabaseException(new InvalidFileException());
+
+			Version = reader.Read();
+			Debug.Assert(Version >= 0);
+
+			ChunkSize = reader.ReadLittleEndian<int>();
+			Debug.Assert(ChunkSize > 0);
```

Comment:
```suggestion
			Debug.Assert(ChunkSize >= 0);
```

It's possible for a chunk to have empty data during a scavenge (before the scavenge merge phase)